crates/ibc/src/core/ics26_routing/context.rs

```diff
@@ -110,6 +110,30 @@ impl OnRecvPacketAck {
 pub type ModuleOutputBuilder = HandlerOutputBuilder<(), ModuleEvent>;
 
 pub trait Module: Send + Sync + AsAnyMut + Debug {
+    #[cfg(feature = "val_exec_ctx")]
+    #[allow(clippy::too_many_arguments)]
+    fn on_chan_open_init_validate(
```

Comment:
As a side note: upon splitting the validation and execution parts, it might not be necessary to have all the fields(args) for both. Perhaps a later investigation would be helpful if we could get rid of some. Otherwise, why we don't pass the own `msg` to the these methods instead of flattening the fields? It seems just the `Signer` fields is the one not used anywhere. Is there any reservation for `Signer`?

Reply:
> upon splitting the validation and execution parts, it might not be necessary to have all the fields(args) for both.

I'd be careful removing some field because we "think" they might not be useful. We have to remember that these fields might be used in ways that we don't foresee by future IBC applications. The current list is [based on the standard](https://github.com/cosmos/ibc/blob/main/spec/core/ics-026-routing-module/README.md#onchanopeninit), and I don't think we should arbitrarily restrict future IBC apps by passing less data. Also, intuitively, I could see an app needing to check every field in validation, and storing every field in execution.

> Otherwise, why we don't pass the own msg to the these methods instead of flattening the fields?

Good question. We don't pass the message directly because not all fields come from the message (e.g. the [channel id](https://github.com/cosmos/ibc-rs/blob/d6fc4c965d2756449d2933968648f79b86be2aaf/crates/ibc/src/core/context.rs#L592)). So although the message and callbacks are very similar, they should be thought of as distinct. We could however create a `struct` that encompasses all these fields as opposed to passing them one by one and needing `#[allow(clippy::too_many_arguments)]`.

WDYT?

Reply:
Thank you for the clarifications. 
I also thought of that `struct` creation, but I was concerned that it may end up in multiple structs that might not offer much more than the current design. 
Let's put it off for now and keep it in mind for a later thought :)